server/src/main/java/org/elasticsearch/common/xcontent/XContentHelper.java

```diff
@@ -518,6 +518,24 @@ public static BytesReference toXContent(ToXContent toXContent, XContentType xCon
         }
     }
 
+    public static XContentType xContentTypeMayCompressed(BytesReference bytes) {
```

Comment:
s/xContentTypeMyCompressed/xContentTypeMaybeCompressed/

And can we add some javadoc here as well?

Reply:
@romseygeek Javadoc for `xContentTypeMaybeCompressed ` added.